graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -153,12 +153,17 @@ def _construct_location_stack_entry(location, num_traverses):
 # - type: a GraphQL type object, like String or Integer, describing the type of that output value
 # - optional: boolean, whether the output is part of an optional traversal and
 #             could therefore have a value of null because it did not exist
-class OutputMetadata(namedtuple("OutputMetadata", ("type", "optional"))):
+# - folded: boolean, whether the output is within a fold scope
+class OutputMetadata(namedtuple("OutputMetadata", ("type", "optional", "folded"))):
```

Comment:
Consider adding an explicit definition + test of whether outputting `_x_count` (by definition inside a `@fold` scope) counts as a "folded" output.

Also consider adding a sanity check on the `type` field, since `folded=True` implies some pretty strong guarantees about the valid values of `type`. The answer here in part depends on the answer to the above.